test/TestProjects/ExactMatchFlattenInheritance/Generated/Extensions/ResourceGroupExtensions.cs

```diff
@@ -86,15 +85,12 @@ public static Pageable<AzureResourceFlattenModel2> GetAzureResourceFlattenModel2
         /// <param name="name"> The String to use. </param>
         /// <param name="parameters"> The AzureResourceFlattenModel2 to use. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        /// <exception cref="ArgumentException"> <paramref name="name"/> is empty. </exception>
-        /// <exception cref="ArgumentNullException"> <paramref name="name"/> or <paramref name="parameters"/> is null. </exception>
+        /// <exception cref="System.ArgumentException"> <paramref name="name"/> is an empty string, and was expected to be non-empty. </exception>
+        /// <exception cref="System.ArgumentNullException"> <paramref name="name"/> or <paramref name="parameters"/> is null. </exception>
         public async static Task<Response<AzureResourceFlattenModel2>> PutAzureResourceFlattenModel2Async(this ResourceGroup resourceGroup, string name, AzureResourceFlattenModel2 parameters, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrEmpty(name, nameof(name));
-            if (parameters == null)
-            {
-                throw new ArgumentNullException(nameof(parameters));
-            }
+            Argument.AssertNotNull(parameters, nameof(parameters));
```

Comment:
Is `parameters` a path parameter?  What causes us to validate it here?

Reply:
This is mgmt logic. I have unified their code with ours (cause now we are doing mostly the same thing), hence the `if` block that is replaced with  `Argument.AssertNotNull`.
@m-nash , what are the validation rules here?

Reply:
If we find issues where management plane isn't in compliance with the guidelines, it would be good to file issues just so we can keep track of these for later.

Reply:
Parameters is a body parameter in this case.  I would imagine it validates because its marked as required.

Reply:
We recently updated the guidelines to make it more clear that we don't validate service parameters: https://azure.github.io/azure-sdk/general_implementation.html#parameter-validation

This is even if they are required - the service should be responsible for reporting that error.  I don't think it's urgent that we fix this, but we should have an issue to track it.

Reply:
For methods like PUT or PATCH this doesn't make much sense.  If we allowed a customer to pass in null what would they expect to have happen?  We literally got this question from a customer who was confused why we would allow it.

It also opens up the issue of the [RestOperations ](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Administration/src/Generated/RoleAssignmentsRestClient.cs#L164)today validate this as well not just the public methods here, and if we remove that then we will get [null reference exceptions](https://github.com/Azure/autorest.csharp/blob/feature%2Fv3/src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs#L30) in the serializer that we will need to fix and test that layer especially against already GAd data plane libraries

Reply:
Agree with Michael.
We don't need to validate the service data, but if the body parameter is required why don't we throw NULL exception directly?
My understanding is the validation should not be happened on the service data itself, which means the value of body content, properties, but should still check shell.